### PR TITLE
adding the required feature in the script as last_del

### PR DESCRIPTION
**Pull request by AutomationBoi**
fixing issue number #1
kindly test the code and let me know the changes required
it will help if you can attach some error screenshots
Regards

### CODE REVIEW

File: dirjump [hunk 1]
```diff
@@ -25,9 +25,23 @@ alias 0="$dirjump_command 10" # 50% fewer keystrokes, why not?
 alias 10='echo "dirjump: use 0 to jump to the 10th directory"'
 
 directory_list=~/.config/dirjump/directory_list
-
+last_dir_remove=~/.config/dirjump/last_dir_remove
 # Get dir path of nth line of directory_list
 # Source: https://stackoverflow.com/a/27056916/9157799
```
Comment (imambungo) [inline dirjump]: These comments should be placed above `get_nth_path()` function.

File: dirjump [hunk 2]
```diff
@@ -111,12 +125,17 @@ dirjump()
 	else
 		echo "dirjump: the requested dir path number is out of range"
 	fi
+	if [[ -f "$last_del" ]]
```
Comment (imambungo) [inline dirjump]: Should be `$last_dir_remove`.

File: dirjump [hunk 3]
```diff
@@ -111,12 +125,17 @@ dirjump()
...
+	then
+        last_del
+        fi
```
Comment (imambungo) [inline dirjump]: These 4 lines should be placed below `print_directory_history` (line 108) to make it works.

And a little indentation fix:
```suggestion
      fi
```

File: dirjump [hunk 4]
```diff
@@ -25,9 +25,23 @@ alias 0="$dirjump_command 10" # 50% fewer keystrokes, why not?
...
 directory_list=~/.config/dirjump/directory_list
-
+last_dir_remove=~/.config/dirjump/last_dir_remove
 # Get dir path of nth line of directory_list
 # Source: https://stackoverflow.com/a/27056916/9157799
+function last_del()
+{
+        time_del=`cat "$last_dir_remove"`
+        curr_time=`date +%s`
+        delta_hour=`echo "($curr_time-$time_del)/3600|bc"`
```
Comment (imambungo) [inline dirjump]:
```suggestion
        delta_hour=`echo "($curr_time-$time_del)/3600" | bc`
```

File: dirjump [hunk 5]
```diff
@@ -25,9 +25,23 @@ alias 0="$dirjump_command 10" # 50% fewer keystrokes, why not?
...
+        if [[ "$delta_hour" -gt 1 ]]
+        then
+        echo "A path was last removed $delta_hour hours ago."
```
Comment (imambungo) [inline dirjump]: Add indentation to improve readability.
```suggestion
                echo "A path was last removed $delta_hour hours ago."
```

File: dirjump [hunk 6]
```diff
@@ -25,9 +25,23 @@ alias 0="$dirjump_command 10" # 50% fewer keystrokes, why not?
...
+        else
+        echo "A path was last removed $delta_hour seconds ago."
```
Comment (imambungo) [inline dirjump]:
```suggestion
                echo "A path was last removed $delta_hour seconds ago."
```

**Review by imambungo — CHANGES_REQUESTED** [review]

File: dirjump [hunk 7]
```diff
@@ -117,6 +150,7 @@ dirjump()
 delete_a_dir_path()
 {
 	grep -Fxv "$1" "$directory_list" > temp; mv temp "$directory_list"
+	date +%s > "$last_dir_remove"
```
Comment (imambungo) [inline dirjump]: This line should be moved to somewhere else. The function `delete_a_dir_path` was originally purposed to [prevent duplicate paths](https://github.com/AutomationBoi/dirjump/blob/ff74ae408a3ee0600c5cc03ff0b5117ab52ed7e2/dirjump#L164) when a  path (already in the list) is added to the top of the list. A path is really removed from the directory history when the list is full and a new path (not in the list yet) is added to the top of the list.

**Review by imambungo — CHANGES_REQUESTED** [review]

File: dirjump (hunk 7 above)
Comment (imambungo) [reply dirjump]: image

In the above image, it says a path was last removed just now, but actually nothing was removed. It just moves `~/.config` from 4 to 1.

image

And when a path was really removed (`~/Projects/Bash/termsystor`), `last_dir_remove` didn't get updated.

**Comment by AutomationBoi** [comment]
looking into it now

**Comment by AutomationBoi** [comment]
i believe apply_max_limit_to_history is the function responsible for deletion of a path can you confirm my analysis.

**Comment by imambungo** [comment]
Yes, you're right. Btw I think we can use number_of_dir_paths() like this:
```
if number_of_dir_paths > history_size
    update last_dir_remove
```

**Comment by AutomationBoi** [comment]
working on the suggestion i am adding this check in the function propose_dir_path after the calling of 
insert_dir_path_to_top creating a push based on this please test

**Comment by imambungo** [comment]
LGTM. Gonna merge this, thanks for the PR :+1:

**Review by imambungo — APPROVED** [review]

**Comment by AutomationBoi** [comment]
:) your welcome

**Comment by AhmadAgus2010** [comment]
:+1:

**Pull request merged**